wallet-connect/src/client/socket.rs

```diff
@@ -133,14 +139,49 @@ impl Socket {
         };
         drop(session);
         self.send_socket_msg(context, id, message)?;
-        let response = rx.await?;
-        let code = response["code"].as_i64();
-        if let Some(value) = code {
-            if -32000 == value {
-                return Err(eyre!("{}", serde_json::to_string(&response)?));
+        // Wrap the future with a `Timeout` set to expire in `pending_requests_timeout` milliseconds.
+        match timeout(
+            Duration::from_millis(context.0.pending_requests_timeout),
+            rx,
+        )
+        .await
+        {
+            Ok(v) => {
+                let response = v?;
+                let code = response["code"].as_i64();
+                if let Some(value) = code {
+                    if -32000 == value {
+                        return Err(eyre!("{}", serde_json::to_string(&response)?));
+                    }
+                }
+                serde_json::from_value(response).wrap_err("failed to parse response")
+            }
+            Err(_) => {
```

Comment:
It is good to print out the error for user.

```
sig1: Err(Eyre({"code":-32000,"data":null,"message":"Failed or Rejected Request"}

Location:
    /Users/damonchen/Documents/play-cpp-sdk/wallet-connect/src/client/socket.rs:154:36))
sig1: Err(Eyre({"code":-32000,"payload":{"reason":"Request is dropped because of timeout","timeout":60000}}

```